marklogic-data-hub/src/test/ml-modules/root/test/suites/data-hub/5/flow/testTargetCollectionAdditivity/flowWithTargetCollectionsAdditivity.sjs

```diff
@@ -0,0 +1,31 @@
+const test = require("/test/test-helper.xqy");
+const hubTest = require("/test/data-hub-test-helper.sjs");
+const flowRunner = require("/data-hub/5/flow/flowRunner.sjs");
+
+let assertions = [];
+let results, collections;
+
+const options = {
+  sourceDatabase: "data-hub-FINAL",
+  targetDatabase: "data-hub-FINAL",
+  targetCollectionsAdditivity: true,
+  outputFormat: 'json'
+};
+
+const content = [{uri:"/customer1.json",
+  value:{"hello": "world"},
+  context: {originalCollections: ["source"]}
+}];
+results = flowRunner.processContentWithFlow("simpleMappingFlow", content, "1", options);
+collections = hubTest.getRecord("/customer1.json").collections;
+
+assertions.push(
+  // doc should have been written
+  test.assertEqual("1", results.lastCompletedStep),
+
+  // doc should have both collection from mapping step and collection from source doc
+  test.assertEqual(2, collections.length),
+  test.assertTrue(collections.includes("Customer")),
```

Comment:
getRecord will alphabetize the collections, and we'd rather use an assertion message here, so:

    test.assertEqual("Customer", collections[0]),
    test.assertEqual("source", collections[1], "The existing collection should have been retained since targetCollectionsAdditivity=true")

Reply:
I think it makes the code more readable to say `collections.includes("Customer")` than `collections[0] == "Customer"`.  Unless you are majorly against this, I'd rather keep the `.includes` and just add a message instead